### PR TITLE
Feature/kas 2670 ingediend door procedurestap zonder ministers decreten vp

### PR DESCRIPTION
**Pull request by sergiofenoll**
Don't display "Nog niet aangevraagd" and "Er zijn nog geen ministers toegevoegd." message for subcases of type "bekrachtiging Vlaamse Regering". Instead, respectively leave the field empty and display "Niet van toepassing".

image

### CODE REVIEW

File: app/pods/cases/case/subcases/subcase/overview/route.js [hunk 1]
```diff
@@ -18,6 +18,7 @@ export default class CasesCaseSubcasesSubcaseOverviewRoute extends Route {
 
   async model(params) {
     const subcase = this.modelFor('cases.case.subcases.subcase');
+    const type = await subcase.type;
```
Comment (erikap) [inline app/pods/cases/case/subcases/subcase/overview/route.js]: By 'decoupling' the type from the subcase here, we loose the relation with subcase. Consequently if the user modifies the type of the subcase, this value doesn't get updated. 
Can be resolved by refreshing the model hook, but that's a lot of data reloading for nothing imo. I'm more inclined to use the relationship in the getter so we make use of the benefits of ember-data: `return [CONSTANTS.SUBCASE_TYPES.BEKRACHTIGING].includes(this.model.type?.get('uri'));`

**Review by erikap — CHANGES_REQUESTED** [review]

File: app/pods/cases/case/subcases/subcase/overview/route.js (hunk 1 above)
Comment (sergiofenoll) [reply app/pods/cases/case/subcases/subcase/overview/route.js]: thanks, didn't take the editing into account.

File: app/models/subcase-type.js [hunk 2]
```diff
@@ -1,12 +1,10 @@
...
-// eslint-disable-next-line ember/no-classic-classes
-export default Model.extend({
-  label: attr('string'),
-  altLabel: attr('string'),
-  scopeNote: attr('string'),
-  subcases: hasMany('subcase', {
-    inverse: null,
-  }),
-});
+export default class Subcase extends Model {
```
Comment (erikap) [inline app/models/subcase-type.js]: Model name is incorrect

File: app/pods/cases/case/subcases/subcase/overview/controller.js [hunk 3]
```diff
@@ -16,6 +17,10 @@ export default class CasesCaseSubcasesSubcaseOverviewController extends Controll
   @tracked governmentAreas;
   @tracked siblingSubcasesCount;
 
+  get showMandateesNotApplicableMessage() {
+    return [CONSTANTS.SUBCASE_TYPES.BEKRACHTIGING].includes(this.model.subcase.get('type')?.get('uri'));
```
Comment (erikap) [inline app/pods/cases/case/subcases/subcase/overview/controller.js]: Is the `get()` for `'type'` required?  `this.model.subcase` is not a proxy object, isn't it?

**Review by erikap — CHANGES_REQUESTED** [review]

File: app/models/subcase-type.js (hunk 2 above)
Comment (sergiofenoll) [reply app/models/subcase-type.js]: Darn, I need to pay more attention :facepalm:

File: app/pods/cases/case/subcases/subcase/overview/controller.js (hunk 3 above)
Comment (sergiofenoll) [reply app/pods/cases/case/subcases/subcase/overview/controller.js]: You're right, not needed here. The semantics when it comes to proxy objects still confuse me a bit :woozy_face:

**Review by erikap — APPROVED** [review]

**Pull request merged**